rpc/src/v1/helpers/light_fetch.rs

```diff
@@ -308,11 +309,11 @@ impl LightFetch {
 			Some(OnDemandResponse::Receipts(b)) => b,
 			_ => panic!(WRONG_RESPONSE_AMOUNT_TYPE_PROOF),
 		}))
+
+	
 	}
 
-	/// Get transaction logs
-	pub fn logs(&self, filter: EthcoreFilter) -> impl Future<Item = Vec<Log>, Error = Error> + Send {
-		use std::collections::BTreeMap;
+	fn logs_common(&self, filter: EthcoreFilter) -> impl Future<Item = BTreeMap<(u64,usize),Log>, Error = Error> + Send {
```

Comment:
There is a slight difference between both logs method, the collect from the second one is different, so I probably did it to avoid a vec iteration (which may be optimize anyway, I do not know).

Reply:
Hrm, do we use a `BTreeMap` because we want the logs sorted by (block_num, block_index)? Then it makes sense to me at least!

Still both `fn logs` and `fn logs_light` collect the values of BTreeMap into a Vec so I would guess it would end up with the same number allocations as before (don't tx hashes retrieval)?

So why not return a Vec<Log> instead or am I missing something?

Reply:
Yes I think that is the purpose.
It is the same allocation in both case, just that with the btree I avoid running an iteration on the vec to create the block query futures.

Reply:
> I avoid running an iteration on the vec to create the block query futures

Not sure what you mean by that, we have created these futures in `BTreeMap` already.
I meant why don't we call `logs_no_tx_hash` from `logs` (no need for `logs_common`)? I see now we're allocating the same `Vec` into `result` in `logs`. And iterating over a `Vec` should be faster, since it's a sequential memory read.